test/functionals/api/v2/containers_controller_test.rb

```diff
@@ -59,9 +61,49 @@ class ContainersControllerTest < ActionController::TestCase
         end
 
         test 'creates a container with correct params' do
+          repository_name = "centos"
+          tag = "7"
+          name = "foo"
+          registry_uri = URI.parse(@registry.url)
           Service::Containers.any_instance.expects(:pull_image).returns(true)
-          Service::Containers.any_instance.expects(:start_container).returns(true)
-          post :create, :container => { :name => 'foo', :registry_id => 3, :image => 'centos:7' }
+          Service::Containers.any_instance.expects(:start_container)
+                                          .returns(true).with do |container|
+            container.must_be_kind_of(Container)
+            container.repository_name.must_equal(repository_name)
+            container.tag.must_equal(tag)
+            container.compute_resource_id.must_equal(@compute_resource.id)
+            container.name.must_equal(name)
+            container.repository_pull_url.must_include(registry_uri.host)
+            container.repository_pull_url.must_include("#{repository_name}:#{tag}")
+          end
+          post :create, :container => { :compute_resource_id => @compute_resource.id,
+                                        :name => name, :registry_id => @registry.id,
+                                        :repository_name => repository_name, :tag => tag  }
+          assert_response :created
+        end
+
+        test 'creates a katello container with correct params' do
+          DockerContainerWizardStates::Image.class_eval do
+            attr_accessor :capsule_id
+          end
+
+          repository_name = "katello_centos"
+          tag = "7"
+          name = "foo"
+          capsule_id = "10000"
+          Service::Containers.any_instance.expects(:start_container!)
+                                          .returns(@container).with do |wizard_state|
+            wizard_state.must_be_kind_of(DockerContainerWizardState)
+            wizard_state.container_attributes[:repository_name].must_equal(repository_name)
+            wizard_state.container_attributes[:tag].must_equal(tag)
+            wizard_state.container_attributes[:compute_resource_id]
+                                             .must_equal(@compute_resource.id)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 33) spaces for indenting an expression spanning multiple lines.